src/order_book.cpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\order_book.h"
+
+#include <cmath>
+#include <memory>
+using std::shared_ptr;
+
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\order.h"
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\transactions.h" 
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\limit_order.h"
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\order_tree.h"
+
+/* Return buy tree */
+OrderTree& OrderBook::get_buy_tree() {
+	return buy_tree_;
+}
+
+/* Return sell tree */
+OrderTree& OrderBook::get_sell_tree() {
+	return sell_tree_;
+}
+
+/* Execute incoming_order or store if it's not completelty filled */
+Transactions OrderBook::execute_order(const std::shared_ptr<Order>& incoming_order) {
```

Comment:
I think `processOrder` will be a better name because execute means that you are running multiple commands. However, here you are only processing one input record.